examples/python/VideoAPI.ipynb

```diff
@@ -0,0 +1,672 @@
+   "outputs": [],
+   "source": [
+    "class RandomDataset(torch.utils.data.IterableDataset):\n",
+    "    def __init__(self, root, epoch_size=None, frame_transform=None, video_transform=None, clip_len=16):\n",
+    "        super(RandomDataset).__init__()\n",
+    "        \n",
+    "        self.samples = get_samples(root)\n",
+    "         \n",
+    "        # allow for temporal jittering\n",
+    "        if epoch_size is None:\n",
+    "            epoch_size = len(self.samples)\n",
+    "        self.epoch_size = epoch_size\n",
+    "        \n",
+    "        self.clip_len = clip_len  # length of a clip in frames\n",
+    "        self.frame_transform = frame_transform  # transform for every frame individually\n",
+    "        self.video_transform = video_transform # transform on a video sequence\n",
+    "\n",
+    "    def __iter__(self):\n",
+    "        for i in range(self.epoch_size):\n",
+    "            # get random sample\n",
+    "            path, target = random.choice(self.samples)\n",
+    "            # get video object\n",
+    "            vid = torch.classes.torchvision.Video(path, \"video\")\n",
+    "            metadata = vid.get_metadata()\n",
+    "            video_frames = [] # video frame buffer \n",
+    "            # seek and return frames\n",
+    "            \n",
+    "            max_seek = metadata[\"video\"]['duration'][0] - (self.clip_len / metadata[\"video\"]['fps'][0])\n",
+    "            start = random.uniform(0., max_seek)\n",
+    "            vid.seek(start)\n",
+    "            while len(video_frames) < self.clip_len:\n",
+    "                frame, current_pts = vid.next()\n",
+    "                video_frames.append(self.frame_transform(frame))\n",
+    "            # stack it into a tensor\n",
+    "            video = torch.stack(video_frames, 0)\n",
+    "            if self.video_transform:\n",
+    "                video = self.video_transform(video)\n",
+    "            output = {\n",
+    "                'path': path,\n",
+    "                'video': video,\n",
+    "                'target': target,\n",
+    "                'start': start,\n",
+    "                'end': current_pts}\n",
+    "            yield output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Given a path of videos in a folder structure, i.e:\n",
+    "```\n",
+    "dataset:\n",
+    "    -class 1:\n",
+    "        file 0\n",
+    "        file 1\n",
+    "        ...\n",
+    "    - class 2:\n",
+    "        file 0\n",
+    "        file 1\n",
+    "        ...\n",
+    "    - ...\n",
+    "```\n",
+    "We can generate a dataloader and test the dataset. \n",
+    "            "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from torchvision import transforms as t\n",
+    "transforms = [t.ToPILImage(), t.Resize((112, 112), interpolation=2), t.ToTensor()]\n",
```

Comment:
This can now be simplified with just `t.Resize((112, 112))` now that we support tensor as input to video ops